tests/test_bucketlist.py

```diff
@@ -0,0 +1,380 @@
+
+        self.data = {
+            "name":"Update work goals",
+            "owner_id": 1
+        }
+
+        response = self.client().put("/api/v1.0/bucketlists/1", data=self.data,
+                                     headers=dict(Authorization="Bearer " + access_token))
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("Bucketlist desscription missing", str(response.data))
+    
+    def test_cannot_update_bucketlist_when_name_is_invalid_format(self):
+        """
+        Test API cannot update a bucketlist if bucketlist is missing description
+        (PUT request)
+        """
+        self.register_user()
+        result = self.login_user()
+        access_token = json.loads(result.data.decode())['access_token']
+        response = self.client().post('/api/v1.0/bucketlists/',
+                                      data=self.bucketlist,
+                                      headers=dict(Authorization="Bearer " + access_token))
+
+        self.data = {
+            "name":"&&(&)((&&(&",
+            "description":"Things To achieve at work(updated)",
+            "owner_id": 1
+        }
+
+        response = self.client().put("/api/v1.0/bucketlists/1", data=self.data,
+                                     headers=dict(Authorization="Bearer " + access_token))
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("Sorry Invalid name format. please put a valid name", str(response.data))
+
+    def test_deletion_of_bucketlist(self):
+        """
+        Test API can delete a bucketlist
+        (DELETE request)
+        """
+        self.register_user()
+        result = self.login_user()
+        access_token = json.loads(result.data.decode())['access_token']
+
+        self.data = {
+            "name":"Update work goals",
+            "description":"Things To achieve at work(updated)",
+            "owner_id": 1
+        }
+
+        response = self.client().post('/api/v1.0/bucketlists/', data=self.data,
+                                      headers=dict(Authorization="Bearer " + access_token))
+
+        response2 = self.client().delete("/api/v1.0/bucketlists/1",
+                                         headers=dict(Authorization="Bearer " + access_token))
+        self.assertEqual(response2.status_code, 200)
+        self.assertIn("bucketlist 1 deleted successfully", str(response2.data))
+
+    def test_deletion_of_nonexistant_bucketlist(self):
+        """
+        Test API cannot delete a bucketlist that doesn't exist (DELETE request)
+        """
+        self.register_user()
+        result = self.login_user()
+        access_token = json.loads(result.data.decode())['access_token']
+
+        response = self.client().delete("/api/v1.0/bucketlists/290834",
+                                        headers=dict(Authorization="Bearer " + access_token))
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("404 Not Found", str(response.data))
```

Comment:
In the base test class register and login a user to get the access token instead of doing it in every test. This will make your work neater and readable.